Refactor debug process

rclee opened the pull request:
Some code to move and remove. This should bring down the size of our binary.

rclee committed:
Remove old debug code and fix some typos

alfwatt reviewed (commented):
Is there a corresponding change in the example app to keep these reports or are we just removing them?

rclee commented:
Example app is next. Working on adding in the hook for that to work now.

rclee committed:
Remove remaining debug code

rclee committed:
Add debug handler and briefs

Handler should be used by our example application to visualize debug events in a more useful way.

rclee committed:
Add ifdef DEBUG

rclee committed:
Add other ifdef

rclee committed:
More DEBUG filtering

rclee committed:
Derp

rclee committed:
Change from #ifdef to #if

#ifdef will only check if DEBUG is being defined and not the value.

alfwatt reviewed (approved):
#ifdef  CODE_WORKING
GitHubCommit()
#endif
